### PR TITLE
Update to latest versions of dependencies

### PR DESCRIPTION
**Pull request by samanpa**
CC @ehuss

### CODE REVIEW

**Comment by ehuss** [comment]
It looks like there are some CI errors that are a little worrisome.  Maybe something changed in semver?  I think Cargo has been hanging back on 0.10 because there were some changes in 0.11 that cause problems.  Those version reqs don't look correct.  

Also, I'm a little uncomfortable with the awkwardness of converting the `metadata.target_directory`. Is there maybe a nicer way to convert that?  Is there a specific reason to update any of these packages?

File: reg-index/src/metadata.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ pub(crate) fn metadata_reg(
         }
         None => cargo_package(
             &actual_manifest_path,
-            &metadata.target_directory,
+            &Path::new(&metadata.target_directory.into_string()),
```
Comment (ehuss) [inline reg-index/src/metadata.rs:119]: It looks like this should work.

```suggestion
            metadata.target_directory.as_ref(),
```

**Comment by samanpa** [comment]
Thanks will take a look at the CI failures again. It looked like stable worked but nightly failed.

**Comment by samanpa** [comment]
> Thanks, will take a look at the CI failures again. It looked like stable worked but nightly failed.

Not sure why this is the case but on nightly the version req looks like 
```
"req":">=0.1.0, <0.2.0"
```

but we are expecting
```
"req":"\^0\.1"
```

**Comment by samanpa** [comment]
Downgrading to semver 0.10.0 fixes this.

**Review by ehuss — APPROVED** [review]
Thanks!

**Pull request merged**